lib/appsignal/integrations/rake.rb

```diff
@@ -0,0 +1,30 @@
+module Rake
+  class Task
+    alias_method :invoke_without_appsignal, :invoke
+
+    def invoke(*args)
+      if Appsignal.active?
+        invoke_with_appsignal(*args)
+      else
+        invoke_without_appsignal(*args)
+      end
+    end
+
+    def invoke_with_appsignal(*args)
+      invoke_without_appsignal(*args)
+    rescue => error
+      transaction = Appsignal::Transaction.create(
+        SecureRandom.uuid,
+        Appsignal::Transaction::BACKGROUND_JOB,
+        Appsignal::Transaction::GenericRequest.new(
+          :params => args
+        )
+      )
+      transaction.set_action(name)
+      transaction.set_error(error)
+      transaction.complete!
+      sleep 0.5 # Give us some time to flush to the agent
```

Comment:
No, it only adds it to the queue. Normally that's fine, but if the Rake process terminates immediately after this (which is to be expected) the worker thread will be killed too.